packages/git/src/common/git.ts

```diff
@@ -691,12 +703,20 @@ export interface Git extends Disposable {
     show(repository: Repository, uri: string, options?: Git.Options.Show): Promise<string>;
 
     /**
-     * It resolves to an array of configured remotes for the given repository.
+     * It resolves to an array of configured remotes names for the given repository.
      *
-     * @param repository the repository to get the remotes.
+     * @param repository the repository to get the remote names.
      */
     remote(repository: Repository): Promise<string[]>;
 
+    /**
+     * It resolves to an array of configured remote objects for the given Git action.
+     *
+     * @param repository the repository to get the remote objects.
+     * @param options `git remote` command refinements.
+     */
+    remote(repository: Repository, options?: Git.Options.Remote): Promise<RemoteModel[]>;
```

Comment:
`options` cannot be optional here. Otherwise, at call-site we cannot infer the return type.

Reply:
fixed!